tests/shared/drive-constraints.spec.js

```diff
@@ -303,52 +303,219 @@ describe('Shared: DriveConstraints', function() {
 
   describe('.isDriveLargeEnough()', function() {
 
-    it('should return true if the drive size is greater than the image size', function() {
-      const result = constraints.isDriveLargeEnough({
+    beforeEach(function() {
+      this.drive = {
         device: '/dev/disk1',
         name: 'USB Drive',
-        size: 1000000001,
+        size: 1000000000,
         protected: false
-      }, {
-        path: 'rpi.img',
-        size: 1000000000
+      };
+    });
+
+    describe('given the final image size estimation flag is false', function() {
+
+      describe('given the original size is less than the drive size', function() {
+
+        beforeEach(function() {
+          this.image = {
+            path: path.join(__dirname, 'rpi.img'),
+            size: {
+              original: this.drive.size - 1,
+              final: {
+                estimation: false
+              }
+            }
+          };
+        });
+
+        it('should return true if the final size is less than the drive size', function() {
+          this.image.size.final.value = this.drive.size - 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return true if the final size is equal to the drive', function() {
+          this.image.size.final.value = this.drive.size;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return false if the final size is greater than the drive size', function() {
+          this.image.size.final.value = this.drive.size + 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.false;
+        });
+
       });
 
-      m.chai.expect(result).to.be.true;
-    });
+      describe('given the original size is equal to the drive', function() {
+
+        beforeEach(function() {
+          this.image = {
+            path: path.join(__dirname, 'rpi.img'),
+            size: {
+              original: this.drive.size,
+              final: {
+                estimation: false
+              }
+            }
+          };
+        });
+
+        it('should return true if the final size is less than the drive size', function() {
+          this.image.size.final.value = this.drive.size - 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return true if the final size is equal to the drive', function() {
```

Comment:
At least you can rest safe in the knowledge that I'll probably catch any omissions :laughing: